### PR TITLE
GOBModel singleton.

### PR DESCRIPTION
**Pull request by peteradrichem**

### CODE REVIEW

File: gobcore/model/__init__.py [hunk 1]
```diff
@@ -155,52 +154,6 @@ def get_inverse_relations(self):
...
-          try:
-              return gob_model[catalog_name]['collections'][collection_name]
-          except KeyError:
-              return None
-        """
-        deprecation("deprecated: use gob_model[catalog_name]['collections'].get(collection_name) …")
-        collections = self.get_collections(catalog_name)
-        return collections[collection_name] if collections and collection_name in collections else None
-
     def get_collection_by_name(self, collection_name):
```
[the remark this answers is not in the document]
Comment (peteradrichem) [reply gobcore/model/__init__.py:157]: Thanks

File: gobcore/model/relations.py [hunk 2]
```diff
@@ -89,21 +88,23 @@ def _get_relation(name):
...
+        dst_catalog = model.data[dst_catalog_name]
         dst_collection = dst_catalog['collections'][dst_collection_name]
         return {
             "catalog": dst_catalog,
             "catalog_name": dst_catalog_name,
             "collection": dst_collection,
             "collection_name": dst_collection_name
         }
-    except (TypeError, KeyError):
+    # hr.nietnatuurlijkepersonen, hr.rechtspersoon
```
[the remark this answers is not in the document]
Comment (peteradrichem) [reply gobcore/model/relations.py:106]: Agreed

File: gobcore/model/relations.py [hunk 3]
```diff
@@ -121,7 +122,9 @@ def _get_relation_name(src, dst, reference_name):
                 f"{dst['catalog']['abbreviation']}_{dst['collection']['abbreviation']}_" +
                 f"{reference_name}").lower()
         return NameCompressor.compress_name(name)
-    except (TypeError, KeyError):
+    # brk.kadastralesubjecten.heeft_rsin_voor => hr.nietnatuurlijkepersonen
+    # hr.sbiactiviteiten.heeft_als_rechtspersoon => hr.rechtspersoon
+    except TypeError:
```
[the remark this answers is not in the document]
Comment (peteradrichem) [reply gobcore/model/relations.py:127]: `_get_destination()` will return a valid _dict_ or `None` for `dst` and when `dst` is `None` the `name` assignment will return a `TypeError: 'NoneType' object is not subscriptable`.

File: gobcore/model/relations.py [hunk 4]
```diff
@@ -188,7 +191,8 @@ def get_relations(model):
         "description": "GOB Relations",
         "collections": {}
     }
-    for src_catalog_name, src_catalog in model.items():
+    # model.data covers GOBModel initialisation.
+    for src_catalog_name, src_catalog in model.data.items():
```
[the remark this answers is not in the document]
Comment (peteradrichem) [reply gobcore/model/relations.py:195]: Correct.
During `GOBModel` class initialisation the `items()` method will raise a `TypeError: items() missing 1 required positional argument: 'self'`. (The type of `cls` in `__new__` is `abc.ABCMeta`.)

File: gobcore/model/__init__.py [hunk 5]
```diff
@@ -32,100 +26,105 @@ class NoSuchCollectionException(NotInModelException):
...
+
+            # Proces GOBModel.data.
+            cls._load_schemas(cls.data)
+            cls._init_data(cls.data)
+
+            cls._initialised = True
+            cls.__instance = singleton
+        return cls.__instance
+
+    @classmethod
```
[the remark this answers is not in the document]
Comment (peteradrichem) [reply gobcore/model/__init__.py:74]: Yep

File: gobcore/model/relations.py [hunk 6]
```diff
@@ -121,7 +122,9 @@ def _get_relation_name(src, dst, reference_name):
                 f"{dst['catalog']['abbreviation']}_{dst['collection']['abbreviation']}_" +
                 f"{reference_name}").lower()
         return NameCompressor.compress_name(name)
-    except (TypeError, KeyError):
+    # brk.kadastralesubjecten.heeft_rsin_voor => hr.nietnatuurlijkepersonen
```
[the remark this answers is not in the document]
Comment (peteradrichem) [reply gobcore/model/relations.py]: Removed

**Review by roel-devries — APPROVED** [review]
lgtm

**Pull request merged**